code/modules/mob/living/carbon/xenomorph/castes/warrior/abilities_warrior.dm

```diff
@@ -196,17 +203,24 @@
 
 	var/turf/T = X.loc
 	var/turf/temp = X.loc
+	var/empowered = X.empower() //Should it knockdown everyone down its path ?
 
 	for (var/x in 1 to fling_distance)
 		temp = get_step(T, facing)
 		if (!temp)
 			break
+		if(empowered)
```

Comment:
early return :
if(!empowered)
   continue

quick cast:
for(var/mob/living/carbon/human/human in temp)